tv_apps/browser/js/authentication_dialog.js

```diff
@@ -0,0 +1,410 @@
+        this.dialogUsernameAreaFocus.bind(this));
+    this.httpAuthenticationPasswordArea.addEventListener('focus',
+        this.dialogPasswordAreaFocus.bind(this));
+
+    this.httpAuthenticationUsernameClear.addEventListener('mouseup',
+      this.clearUsername.bind(this));
+    this.httpAuthenticationUsernameClear.addEventListener('keyup',
+        this.dialogButtonKeyup.bind(this));
+    this.httpAuthenticationUsernameClear.tabIndex = 0;
+
+    this.httpAuthenticationPasswordClear.addEventListener('mouseup',
+      this.clearPassword.bind(this));
+    this.httpAuthenticationPasswordClear.addEventListener('keyup',
+        this.dialogButtonKeyup.bind(this));
+    this.httpAuthenticationPasswordClear.tabIndex = 0;
+
+    this.httpAuthenticationOk.addEventListener('mouseup', this);
+    this.httpAuthenticationOk.addEventListener('keyup',
+        this.dialogButtonKeyup.bind(this));
+    this.httpAuthenticationOk.tabIndex = 0;
+
+    this.httpAuthenticationCancel.addEventListener('mouseup', this);
+    this.httpAuthenticationCancel.addEventListener('keyup',
+        this.dialogButtonKeyup.bind(this));
+    this.httpAuthenticationCancel.tabIndex = 0;
+  },
+
+  /**
+   * Default event handler for mozbrowser event and click event.
+   * @param {Event} evt
+   * @param {String} origin Tab ID
+   */
+  handleEvent: function ad_handleEvent(evt, origin) {
+    switch (evt.type) {
+      case 'mozbrowserusernameandpasswordrequired':
+        evt.preventDefault();
+        this.currentEvents[origin] = evt;
+
+        // Show authentication dialog only if
+        // the frame is currently displayed.
+        this.show(origin);
+        break;
+
+      case 'mouseup':
+        if (evt.currentTarget === this.httpAuthenticationCancel) {
+          this.focusIndex.x = 2;
+          this.focusIndex.y = 1;
+          this.cancelHandler();
+        } else {
+          this.focusIndex.x = 2;
+          this.focusIndex.y = 0;
+          this.confirmHandler();
+        }
+        break;
+    }
+  },
+
+  /**
+   * Show dialog and set message.
+   * @param {String} origin Tab ID
+   */
+  show: function ad_show(origin) {
+    this.currentOrigin = origin;
+    var evt = this.currentEvents[origin];
+
+    this.httpAuthenticationDialog.classList.remove('hidden');
+
+    // XXX: We don't have a better way to detect login failed.
+    if (this._confirmed[origin]) {
+      //this.httpAuthenticationMessage.setAttribute('data-l10n-id',
+        //'the-username-or-password-is-incorrect');
+      //this.httpAuthenticationMessage.classList.add('error');
+    } else {
+      var msg = navigator.mozL10n.get('WB_LT_AUTH_MESSAGE_1_1')
+              + ' ' + evt.detail.host + ' '
+              + navigator.mozL10n.get('WB_LT_AUTH_MESSAGE_1_2') + '<br>'
+              + navigator.mozL10n.get('WB_LT_AUTH_MESSAGE_2') + ' '
+              + evt.detail.realm;
+      this.httpAuthenticationMessage.innerHTML = msg.trim();
```

Comment:
This should not look like this. Instead it should be sth like:

`WB_LT_AUTH_MESSAGE = The server {{ host }} requires a user name and password.<br/>Message from the server {{ msg }}`

and in JS:

`navigator.mozL10n.setAttributes(this.httpAuthenticationMessage, 'WB_LT_AUTH_MESSAGE', {
  host: evt.detailt.host,
  msg: evt.detail.realm
});`